src/Models/BaseElement.php

```diff
@@ -325,10 +325,20 @@ public function write(
      *
      * {@inheritDoc}
      */
-    public function onBeforeWrite()
+    protected function onBeforeWrite()
     {
         parent::onBeforeWrite();
+        $this->updateSort();
+    }
 
+    /**
+     * Increment the sort order if one hasn't been already defined. This
+     * ensures that new elements are created at the end of the list by default.
+     *
+     * {@inheritDoc}
+     */
+    public function updateSort()
```

Comment:
This method is added so that Resolver.php can call it, previously it was calling onBeforeWrite() direclty